actionpack/test/dispatch/request/session_test.rb

```diff
@@ -205,15 +211,24 @@ def call(env)
       end
 
       def app
-        @app ||= RoutedRackApp.new(Router)
+        @app ||= Rack::Lint.new(RoutedRackApp.new(Router))
       end
 
       def test_session_follows_rack_api_contract_1
-        get "/mysessionapp"
+        get "/mysessionapp", env: { Rack::RACK_SESSION => Session.create(store, ActionDispatch::Request.empty, {}) }
```

Comment:
Ah, I think you ran into something I didn't consider, which is that `rack.session` is only verified on the way in to the request 🤔 

Since this is an integration test it would be best if we test how Rails itself puts the Session in the `env` instead of doing it ourselves. I think we can do that by putting `Rack::Lint` _after_ the session is touched in `MySessionApp` (also we should probably use `ActionDispatch::Request` so we're testing Rails and not Rack):

```
class MySessionApp
  def call(env)
    request = ActionDispatch::Request.new(env)
    request.session["hello"] = "Hello from MySessionApp!"
    Rack::Lint.new(-> { [ 200, {}, ["Hello from MySessionApp!"] ] }).call(env)
  end
end
```

Reply:
This one raises

> ActionDispatch::Request::Session::DisabledSessionError: Your application has sessions disabled. To write to the session you must first configure a session store

Could you please help me figure out what's the best way to fix this? 🤔

Reply:
Fixed in 28ca2b3f9f :bow:

Reply:
Thanks!